dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1849,5 +1785,78 @@ String DeltaMergeStore::getLogTracingId(const DMContext & dm_ctx)
         return fmt::format("table_id={}", physical_table_id);
     }
 }
+
+std::pair<SegmentPtr, bool> DeltaMergeStore::getSegmentByStartKeyInner(const RowKeyValueRef & start_key)
+{
+    std::shared_lock lock(read_write_mutex);
+    auto seg_it = segments.upper_bound(start_key);
+    return seg_it != segments.end() ? std::pair{seg_it->second, segments.empty()} : std::pair{nullptr, segments.empty()};
+}
```

Comment:
seems only called by `DeltaMergeStore::getSegmentByStartKey`, maybe no need to abstract to a new function?